iGlance/iGlance/iGlance/AppDelegate.swift

```diff
@@ -195,4 +195,121 @@ class AppDelegate: NSObject, NSApplicationDelegate {
+        openPanel.nameFieldLabel = "Import path:"
+        openPanel.nameFieldStringValue = "settings.json"
+        openPanel.allowedFileTypes = ["json"]
+        openPanel.isExtensionHidden = false
+        openPanel.canSelectHiddenExtension = false
+        openPanel.runModal()
+
+        guard let importURL = openPanel.url else {
+            return
+        }
+
+        do {
+            let fileContents = try String(contentsOf: importURL, encoding: String.Encoding.utf8)
+            let jsonDecoder = JSONDecoder()
+            let jsonData = fileContents.data(using: .utf8)!
+            let newObject = try jsonDecoder.decode(UserSettings.self, from: jsonData)
+            AppDelegate.userSettings.settings = newObject.settings
+
+            // Set new update interval
+            self.changeUpdateLoopTimeInterval(interval: AppDelegate.userSettings.settings.updateInterval)
+
+            // Clear image cache of bar graphs
+            AppDelegate.menuBarItemManager.cpuUsage.barGraph.clearImageCache()
+            AppDelegate.menuBarItemManager.memoryUsage.barGraph.clearImageCache()
+
+            // Set correct visibility of menubaritems
+            AppDelegate.menuBarItemManager.updateMenuBarItems()
+
+            // Restart main window
+            mainWindow.close()
+            mainWindow = NSStoryboard(name: "Main", bundle: nil).instantiateController(withIdentifier: "MainWindowController") as! MainWindowController
+            sleep(1)
+            showMainWindow()
+        } catch {
+            let errorAlert = NSAlert()
+            errorAlert.messageText = "Error"
+            errorAlert.informativeText = "An error occured while importing settings"
+            errorAlert.alertStyle = .critical
+            errorAlert.addButton(withTitle: "OK")
+            errorAlert.runModal()
+        }
+    }
+
+    @IBAction private func resetSettings(sender: AnyObject) {
+        let confirm = NSAlert()
+        confirm.messageText = "Reset Settings"
+        confirm.informativeText = "Are you sure that you want to reset all settings?"
+        confirm.alertStyle = .warning
+        confirm.addButton(withTitle: "Yes")
+        confirm.addButton(withTitle: "Cancel")
+        let res = confirm.runModal()
+
+        if res == .alertSecondButtonReturn {
+            return
+        }
+
+        let defaultSettings = UserSettings(isDefault: true)
+        AppDelegate.userSettings.settings = defaultSettings.settings
+
+        // Set new update interval
+        self.changeUpdateLoopTimeInterval(interval: AppDelegate.userSettings.settings.updateInterval)
+
+        // Clear image cache of bar graphs
+        AppDelegate.menuBarItemManager.cpuUsage.barGraph.clearImageCache()
+        AppDelegate.menuBarItemManager.memoryUsage.barGraph.clearImageCache()
+
+        // Set correct visibility of menubaritems
+        AppDelegate.menuBarItemManager.updateMenuBarItems()
+
+        // Restart main window
+        mainWindow.close()
```

Comment:
Why do we have to close and re-instantiate the whole window? Shouldn't this be handled by the `updateGUIComponents` functions to update the GUI?

We can access the currently displayed subclass of the `MainViewViewController` with this code:
```swift
guard let mainWindowVC = mainWindow.contentViewController as? MainWindowViewController else {
    DDLogError("Could not call the 'contentViewController' of the main window to 'MainWindowViewController'")
    return
}
guard let contentManagerVC = mainWindowVC.contentManagerViewController else {
    DDLogError("'contentManagerViewController' of main window is nil")
    return
}
guard let mainViewVC = contentManagerVC.currentViewController as? MainViewViewController else {
    DDLogError("Could not cast the 'currentViewController' of the 'ContentManagerViewController' to 'MainViewViewController'")
    return
}
mainViewVC.updateGUIComponents()
```

Maybe we can add some kind of `updateGUIofCurrentMainView`-function to the private functions above and call it here and in line 270.